src/chain.h

```diff
@@ -465,8 +468,8 @@ class CChain
             return nullptr;
     }
 
-    /** Return the maximal height in the chain. Is equal to chain.Tip() ? chain.Tip()->nHeight : -1. */
-    int Height() const { return vChain.size() - 1; }
+    /** Return the maximal height in the chain. */
+    int Height() const { return tip.load() ? tip.load()->nHeight : -1; }
     /** Set/initialize a chain with a given tip. */
```

Comment:
I take it we are now lock free here because tip is atomic...maybe a comment to that effect?

+/** Return the maximal height in the chain.  Does not require cs_main  */

Reply:
ok, also added the same note for Tip()